src/category_theory/abelian/additive_functor.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2021 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+
+import category_theory.preadditive
+import category_theory.abelian.basic
+
+/-!
+# Additive Functors
+
+A functor between two preadditive categories is called *additive*
+provided that the induced map on hom types is a morphism of abelian
+groups.
+
+# Implementation details
+
+`functor.additive` is a `Prop`-valued class, defined by saying that
+for every two objects `X` and `Y`, the map
+`F.map : (X ⟶ Y) → (F.obj X ⟶ F.obj Y)` is a morphism of abelian
+groups.
+
+# Projects (in the case of abelian categories):
+
+- Prove that an additive functor preserves finite biproducts
+- Prove that a functor is additive if it preserves finite biproducts
+-/
+
+namespace category_theory
+
+/-- A functor `F` is additive provided `F.map` is an additive homomorphism. -/
+class functor.additive {C D : Type*} [category C] [category D]
+  [preadditive C] [preadditive D] (F : C ⥤ D) : Prop :=
+(map_zero' : Π {X Y : C}, F.map (0 : X ⟶ Y) = 0 . obviously)
+(map_add' : Π {X Y : C} {f g : X ⟶ Y}, F.map (f + g) = F.map f + F.map g . obviously)
+
+restate_axiom functor.additive.map_zero'
+restate_axiom functor.additive.map_add'
+
+attribute [simp] functor.additive.map_zero functor.additive.map_add
+
+section preadditive
+
+variables {C D : Type*} [category C] [category D] [preadditive C]
+  [preadditive D] (F : C ⥤ D) [functor.additive F]
+
+namespace functor
+
+/-- `F.add_map` is an additive homomorphism whose underlying function is `F.map`. -/
+@[simps]
+def add_map {X Y : C} : (X ⟶ Y) →+ (F.obj X ⟶ F.obj Y) :=
+{ to_fun := λ f, F.map f,
+  map_zero' := additive.map_zero,
+  map_add' := λ _ _, additive.map_add }
+
+lemma add_map_spec {X Y : C} {f : X ⟶ Y} : F.add_map f = F.map f := rfl
+
+lemma add_map_spec' {X Y : C} : ⇑(F.add_map : (X ⟶ Y) →+ _) = @map C _ D _ F X Y := rfl
+
+@[simp]
+lemma additive.map_neg {X Y : C} {f : X ⟶ Y} : F.map (-f) = - F.map f :=
```

Comment:
This and the next lemma should be in the functor namespace so we can use for Boston.

Reply:
I put these in the `functor.additive` to match `functor.additive.map_zero` and `functor.additive.map_add` which are generated by `restate_axiom`. 

Are you sure you want these to be `functor.map_neg` etc.?

Reply:
Certainly they should all be in the same place, I agree. The question is whether we want to be able to write `F.map_neg`, etc., in which case they all need to be in the functor namespace. (Sorry, my comment above got mangled by autocorrect.) I don't think this is essential, so I'm happy either way.

Reply:
Okay. Since `F.map f` is defeq to `F.map_add_hom f`, we can always use `F.map_add_hom.map_neg`, etc. 
So I think I will leave it at is for now.

Reply:
Defeq isn't enough for rw too not know the difference though